File: yanwte-core/src/test/java/integrationTests/mixedTest/extspace2/TestExtension2.java
package integrationTests.mixedTest.extspace2;

import integrationTests.mixedTest.TestData;
import integrationTests.mixedTest.TestDataExt;
import integrationTests.mixedTest.TestExtensionPoint;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * @author Winter Young
 * @since 2016/1/27
 */
public class TestExtension2 implements TestExtensionPoint {
    @Nullable
    @Override
    public String foo(@NotNull TestData testData) {
        TestDataExt ext = (TestDataExt) testData.getDataExtension("integrationTests.mixedTest.extspace1");
        assert ext != null;
        int i = ext.getI();
        if (i % 2 != 0) {
            return i - 1 + "";
        }
        return null;
    }
}

File: yanwte-core/src/test/java/integrationTests/mixedTest/extspace2/TestExtension2.kt
package integrationTests.mixedTest.extspace2

import integrationTests.mixedTest.TestData
import integrationTests.mixedTest.TestDataExt
import integrationTests.mixedTest.TestExtensionPoint

class TestExtension2 : TestExtensionPoint {
    override fun foo(testData: TestData): String? {
        val testDataExt = testData.getDataExtension<TestDataExt>("integrationTests.mixedTest.extspace1")!!
        return testDataExt.i.let { i ->
            if (i % 2 != 0) i - 1 else null
        }?.let {
            it.toString()
        }
    }
}